internal/engine/fake_notify_test.go

```diff
@@ -0,0 +1,124 @@
+package engine
+
+import (
+	"sync"
+
+	"github.com/windmilleng/tilt/internal/watch"
+)
+
+type fakeMetaWatcher struct {
+	events chan watch.FileEvent
+	errors chan error
+
+	mu         sync.Mutex
+	subs       []chan watch.FileEvent
+	subsErrors []chan error
+}
+
+func newFakeMetaWatcher() *fakeMetaWatcher {
+	r := &fakeMetaWatcher{events: make(chan watch.FileEvent), errors: make(chan error)}
+	go r.loop()
+	return r
+}
+
+func (w *fakeMetaWatcher) newSub() (watch.Notify, error) {
+	subCh := make(chan watch.FileEvent)
+	errorCh := make(chan error)
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	w.subs = append(w.subs, subCh)
+	w.subsErrors = append(w.subsErrors, errorCh)
+	return newFakeWatcher(subCh, errorCh), nil
+}
+
+func (w *fakeMetaWatcher) getSubs() []chan watch.FileEvent {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	return w.subs
```

Comment:
this probably should copy the slice?

Reply:
i think you may have misunderstood my comment? i'll just send you a pr

Reply:
Sounds good!